k2/csrc/cuda/array.h

```diff
@@ -54,29 +55,22 @@ class Array1 {
     // `Eval(ContextPtr, T*, int, Callable&)` now
     Eval(ctx, Data(), size, std::forward<Callable>(callable));
   }
+  */
 
   Array1(ContextPtr ctx, int32_t size) { Init(ctx, size); }
 
   // Creates an array that is not valid, e.g. you cannot call Context() on it.
   // TODO(haowen): why do we need this version?
   Array1() : dim_(0), byte_offset_(0), region_(nullptr) {}
 
-  Array1(int32_t dim, RegionPtr region, int32_t byte_offset)
-      : dim_(dim), region_(region), byte_offset_(byte_offset) {}
-
   Array1(ContextPtr ctx, int32_t size, T elem) {
     Init(ctx, size);
-    T *data = Data();
-    // TODO(haowen): need capture *this, not sure it's valid in constructor
-    /*
-    auto lambda = [=] __host__ __device__(int32_t i) -> void {
-      data[i] = elem;
-    };
-    Eval(ctx, dim_, lambda);
-    */
+    *this = elem;
```

Comment:
well, it's strange that it's OK to call `operator=` to do the same initialization things in the constructor...the code are totally same...